### DISCUSSION TITLE
How to Achieve Streaming Upload of 10MB Chunks Using Asynchronous Chunking?

### DISCUSSION THREAD

**Discussion by Mnoisec** [encode/httpx#2842] [Q&A]
I'm attempting to implement asynchronous chunked uploading for handling file upload tasks. Specifically, I need to break a file into 10MB-sized chunks and wish to upload these chunks to the target server in a streaming fashion for improved efficiency and performance. I've gone through some documentation, but I'm still unclear on how to implement this feature. Could someone provide some guidance or sample code to help me get started with asynchronous chunked streaming upload? Your assistance would be greatly appreciated!

**Comment by lovelydinosaur** [comment]
If you're uploading files using [multipart file uploads](https://www.python-httpx.org/quickstart/#sending-multipart-file-uploads) then the streaming will be handled for you.

If you're uploading [raw binary content](https://www.python-httpx.org/quickstart/#sending-binary-request-data) then use a generator which iteratively returns the data in 10MB sized chunks.

*Is this a sufficiently helpful answer?*
